numpy/core/src/umath/umathmodule.c

```diff
@@ -95,7 +95,10 @@ ufunc_frompyfunc(PyObject *NPY_UNUSED(dummy), PyObject *args, PyObject *kwds) {
 
     pyname = PyObject_GetAttrString(function, "__name__");
     if (pyname) {
-        (void) PyString_AsStringAndSize(pyname, &fname, &fname_len);
+        PyObject *ascii_name = PyUnicode_AsASCIIString(pyname);
+        PyBytes_AsStringAndSize(ascii_name, &fname, &fname_len);
+        Py_DECREF(pyname);
+        pyname = ascii_name;
```

Comment:
I think we should change ufunc names to be unicode (aka, declare the ->name field as utf8), which makes them match python function names.

Reply:
Actually, `ufunc.__name__` already decodes with utf8, so I think this is an outright bug.

Reply:
@eric-wieser Is the suggestion that we should be encoding as utf-8 rather using ascii?

Reply:
Yes, we should just call `PyUnicode_AsUTF8AndSize` rather than creating an intermediate `bytes` object.